src/Cache/CacheOperationTypes/CacheOperationTypes.php

```diff
@@ -980,53 +989,49 @@ public function asError(): CacheDictionaryGetFieldsResponseError|null
     }
 }
 
-class CacheDictionaryGetFieldsResponseSuccess extends CacheDictionaryGetFieldsResponse
+class CacheDictionaryGetFieldsResponseHit extends CacheDictionaryGetFieldsResponse
 {
-    private array $responsesList = [];
+    private array $responses = [];
+    private array $valuesDictionary = [];
 
-    public function __construct(_DictionaryGetResponse $responses = null, ?int $numRequested = null)
+    public function __construct(_DictionaryGetResponse $responses, ?array $fields = null)
     {
-        if (!is_null($responses) && is_null($numRequested)) {
-            parent::__construct();
-            foreach ($responses->getFound()->getItems() as $response) {
-                if ($response->getResult() == ECacheResult::Hit) {
-                    $this->responsesList[] = new CacheDictionaryGetFieldResponseHit(null, $response->getCacheBody());
-                }
-                if ($response->getResult() == ECacheResult::Miss) {
-                    $this->responsesList[] = new CacheDictionaryGetFieldResponseMiss();
-                } else {
-                    $this->responsesList[] = new CacheDictionaryGetFieldResponseError(new UnknownError(strval($response->getResult())));
-                }
-            }
-        }
-        if (is_null($responses) && !is_null($numRequested)) {
-            foreach (range(0, $numRequested - 1) as $ignored) {
-                $this->responsesList[] = new CacheDictionaryGetFieldResponseMiss();
+        $counter = 0;
+        parent::__construct();
```

Comment:
Kind of a nit, but the call to the parent constructor should pretty much always be the first line in a function:
```suggestion
        parent::__construct();
        $counter = 0;
```